ddtrace/appsec/processor.py

```diff
@@ -275,14 +329,28 @@ def on_span_finish(self, span):
             if body is not None:
                 data[_Addresses.SERVER_REQUEST_BODY] = body
 
+        if self._is_needed(_Addresses.HTTP_CLIENT_IP):
+            remote_ip = _context.get_item("http.request.remote_ip", span=span)
+            if remote_ip:
+                data[_Addresses.HTTP_CLIENT_IP] = remote_ip
+
         log.debug("[DDAS-001-00] Executing AppSec In-App WAF with parameters: %s", data)
         ddwaf_result = None
-        try:
-            ddwaf_result = self._ddwaf.run(data, self._waf_timeout)  # res is a serialized json
-        except OSError:
-            log.warning("Error executing Appsec In-App WAF: ", exc_info=True)
-        except Exception as e:
-            log.warning("Error executing Appsec In-App WAF: %s", repr(e))
+        total_runtime = 0.0
+        total_overall_runtime = 0.0
+        blocked_request = _context.get_item("http.request.blocked", span=span)
+        if not blocked_request:
+            try:
+                ddwaf_result = self._ddwaf.run(data, self._waf_timeout)  # res is a serialized json
+                total_runtime = ddwaf_result.runtime
+                total_overall_runtime = ddwaf_result.total_runtime
+            except OSError:
+                log.warning("Error executing Appsec In-App WAF: ", exc_info=True)
+            except Exception as e:
+                log.warning("Error executing Appsec In-App WAF: %s", repr(e))
```

Comment:
no `exc_info` here?

Reply:
Added to the remaining exception handler.